app/controllers/api/comments_controller.rb

```diff
@@ -0,0 +1,26 @@
+class Api::CommentsController < ApplicationController
+  def index
+    user = User.find(params[:user_id])
+    post = user.posts.find(params[:post_id])
+    comments = post.comments
+    render json: comments.to_json(only: %i[id text created_at updated_at comments_counter likes_counter])
+  end
+
+  def create
+    post = Post.find(params[:post_id])
+    comment = post.comments.build(comment_params)
+    comment.author_id = current_user.id
+
+    if comment.save
+      render json: comment, status: :created
+    else
+      render json: { error: comment.errors.full_messages.join(', ') }, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+end
```

Comment:
- [ ] When I try to post a comment to the application, I get this error:

![image](https://github.com/skabeo/blog-app/assets/77106962/8f79a72f-7fa4-4bb2-bf20-df8a4095b0e6)

#### The error is related to **_Cross-Origin Request Forgery (CSRF)_** protection in Rails. `CSRF` protection is a security feature that prevents malicious websites from making unauthorized requests on behalf of a user. 😎 

#### When making a POST request from a web page, Rails expects an authenticity token to be included in the request headers to verify that the request is coming from a trusted source. 💻 

----

#### To resolve this, you can put the following code at the top of your `API::CommentsController` class:

```ruby
skip_before_action :verify_authenticity_token, if: -> { request.format.json? }
```

#### This will skip CSRF protection only for requests with a `JSON` format. By disabling CSRF protection for API endpoints, you allow clients to make requests without needing to provide a CSRF token, which is appropriate for API authentication. 👍 🖱️ 

----

#### N.B: Remember that handling CSRF protection correctly is essential for security, especially when dealing with web development and API endpoints. Always follow best practices and consider the specific use case and security requirements of your application. However, for this application, implementing it this way will suffice. 👍 😎